distribution/push_v2.go

```diff
@@ -263,6 +265,37 @@ func (pd *v2PushDescriptor) Upload(ctx context.Context, progressOutput progress.
 	// then push the blob.
 	bs := pd.repo.Blobs(ctx)
 
+	// attempt to find another repository to mount the layer from
+	for _, blobsum := range possibleBlobsums {
```

Comment:
Changing above comment to:
```// Attempt to find another repository in the same registry to mount the layer from to avoid an unnecessary upload```

Changing ```// remove bad sources``` to:
```// Unable to mount layer from this repository, so this source mapping is no longer valid```